invenio_rdm_records/services/components/internal_notes.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM service component for custom fields."""
+import uuid
+from copy import copy, deepcopy
+from datetime import datetime, timezone
+from invenio_drafts_resources.services.records.components import ServiceComponent
+
+
+class InternalNotesComponent(ServiceComponent):
+    """Service component for custom fields."""
+
+    field = "internal_notes"
+
+    def create(self, identity, data=None, record=None, **kwargs):
+        """Inject note to the record."""
+        notes = data.get(self.field, [])
+        for note in notes:
+            note.update(
+                {
+                    "added_by": {"user": identity.id},
+                    "timestamp": datetime.now(timezone.utc).isoformat(),
+                    "id": str(uuid.uuid4()),
+                }
+            )
+        record.update({"internal_notes": notes})
+
+    def update_draft(self, identity, data=None, record=None, **kwargs):
+        """Inject parsed metadata to the record."""
+        notes_to_update = data.get(self.field, [])
+        notes = deepcopy(record.get(self.field, []))
+        ids_to_check = [note["id"] for note in notes]
+
+        # reverse list to keep proper iterator reference after remove
+        for note in reversed(notes_to_update):
+            note.setdefault("id", str(uuid.uuid4()))
+            if note["id"] in ids_to_check:
+                # exclude existing IDs from incoming data
+                # because we don't modify existing notes
+                notes_to_update.remove(note)
+                ids_to_check.remove(note["id"])
+                continue
```

Comment:
are we simply skipping previously existing notes here? 

I think that it is not very clear the contract of the API. If I understand correctly, we only accept:
1. insert of new notes: the client will only send the notes text (no `id` sent or other fields)
2. deletion
We don't accept updates. In any case, the client should send ALL existing notes.

I feel like it is a bit too complicated the code here, and not easy to capture the constraints here. I suggest checking if there is an easier way to implement it, and add docstring.

Reply:
added docstring as discussed